⚙️ User Mock API 등록

Legitgoons opened the pull request:
## 작업 이유
User API에 대한 Mock API 등록

<br/>

## 작업 사항
- [Wiki API Specification](https://github.com/CollaBu/pennyway-client-webview/wiki/API-Specification) 문서에 업로드 되어있는 API를 기반으로 하였습니다.
- User Interface에 feedCount, followingCount, followerCount를 추가했습니다.
- ProfileFeed Interface와 Relationship Interface를 구현했습니다.

<br/>

## 리뷰어가 중점적으로 확인해야 하는 부분
- [x] ProfileFeed Interface와 Relationship Interface가 추가된 것을 인지했는지
- [x] User Interface에 feedCount, followingCount, followerCount가 추가된 것을 인지했는지
- [x] 구현된 항목들이 API Wiki와 일치하는지
- [x] userHandler가 정상적으로 구현되었는지

<br/>

## 발견한 이슈

Legitgoons committed:
feat: user.d.ts의 User Interface 수정

누락된 feedCount, followingCount, followerCount와 팔로우 확인에 필요한 relationshipStatus를 추가하였습니다.

Legitgoons committed:
feat: users mock data 수정

interface에 추가된 feedCount, followingCount, followerCount, relationshipStatus를 추가

Legitgoons committed:
feat: profileFeed.d.ts에 ProfileFeed Interface 구현

Legitgoons committed:
feat: profileFeeds mock data 구현

Legitgoons committed:
feat: userHandlers 구현

Legitgoons committed:
feat: mocking worker에 userHandlers 추가

Legitgoons committed:
chore: 오타 삭제

Legitgoons committed:
feat: ProfileFeed interface 수정

wiki에 정의된 api에 알맞게 profileImage -> thubnailImage로 수정

Legitgoons committed:
feat: const profileFeeds 수정

profileImage -> thubnailImage로 수정

Legitgoons committed:
feat: Relationship interface 추가

팔로우 확인에 사용되는 relationshipStatus값을 가지는 interface입니다.
wiki의 정의대로 User Interface에서 분리하였습니다.

Legitgoons committed:
feat: relationshipStatus mock data 구현

user mock data 내부에 구현되어 있었으나, interface에 따라 분리하였습니다.

Legitgoons committed:
feat: userHandlers 수정

users에서 relationshipStatus를 가져오던 followInfo를 relationshipStatus에서 가져오도록 수정했습니다.

BangDori commented on src/app/mocks/handler/user.ts:
```diff
@@ -0,0 +1,145 @@
+      currentPageNumber: pageCount,
+      pageSize: formattedPage,
+      numberOfElements: profileFeedsData.length,
+      hasNext,
```
P1: 저희 `hasNext` 말구 `hasNextPage` 로 통일하기루 했습니다!! [피드 목록 조회 API](https://github.com/CollaBu/pennyway-client-webview/wiki/Feed-API-Specification#1%EF%B8%8F%E2%83%A3-%ED%94%BC%EB%93%9C-%EB%AA%A9%EB%A1%9D-%EC%A1%B0%ED%9A%8C) 확인해보시면 hasNextPage로 되어 있습니당!

Legitgoons replied:
넵 수정 완료했습니다!

BangDori commented on src/shared/consts/types/user.d.ts:
```diff
@@ -3,4 +3,12 @@ export interface User {
+  followerCount: number;
+}
+
+export interface Relationship {
```
P2: 저희 Wiki에 등록한 [Type & Interface Rule](https://github.com/CollaBu/pennyway-client-webview/wiki/Type-%26-Interface-Rule) 확인해보시면, 단일 변수의 경우에는 `type`을 사용하기로 했어요!!

`RelationShipStatus`은 `interface`보다 `type`으로 변경하는 게 좋을 것 같아요!

Legitgoons replied:
넵 type으로 수정했습니다!

BangDori reviewed (changes requested):
컨벤션이 맞지 않는 부분이 있어서 이 부분만 수정해주시면 될 것 같습니다! ㅎㅎ

고생많으셨습니다.

Legitgoons committed:
feat: RelationshipStatus union type으로 수정

단일 변수만 있기에 interface에서 union type으로 수정
이름도 Relationship -> RelationshipStatus로 변경

Legitgoons committed:
feat: relationshipStatus mock data 수정

Legitgoons committed:
feat: userHandler내 hasNextPage 변수 이름 수정

BangDori reviewed (approved):
확인했습니다!!~
